packages/backend/src/app/workers/flow-worker/flow-queue-consumer.ts

```diff
@@ -26,50 +27,77 @@ const oneTimeJobConsumer = new Worker<OneTimeJobData, unknown, ApId>(
     },
 )
 
-const repeatableJobConsumer = new Worker<RepeatableJobData, unknown, ApId>(
-    REPEATABLE_JOB_QUEUE,
+const scheduledJobConsumer = new Worker<ScheduledJobData, unknown, ApId>(
+    SCHEDULED_JOB_QUEUE,
     async (job) => {
-        logger.info(`[repeatableJobConsumer] job.id=${job.name} job.type=${job.data.triggerType}`)
-        const { data } = job
+        logger.warn(job.data, '[FlowQueueConsumer#scheduledJobConsumer] job.data')
+        logger.info(`[scheduledJobConsumer] job.name=${job.name}`)
 
-        try {
-            // TODO REMOVE AND FIND PERMANENT SOLUTION
-            const instance = await flowInstanceService.get({
-                projectId: data.projectId,
-                flowId: data.flowVersion.flowId,
-            })
-            if (isNil(instance) || instance.status !== FlowInstanceStatus.ENABLED || instance.flowVersionId !== data.flowVersion.id) {
-                captureException(new Error(`[repeatableJobConsumer] removing job.id=${job.name} instance.flowVersionId=${instance?.flowVersionId} data.flowVersion.id=${data.flowVersion.id}`))
-                await triggerUtils.disable({
-                    projectId: data.projectId,
-                    flowVersion: data.flowVersion,
-                    simulate: false,
-                })
-                return
-            }
-            if (data.triggerType === TriggerType.PIECE) {
-                await consumePieceTrigger(data)
-            }
-        }
-        catch (e) {
-            if (e instanceof ActivepiecesError && e.error.code === ErrorCode.TASK_QUOTA_EXCEEDED) {
-                logger.info(`[repeatableJobConsumer] removing job.id=${job.name} run out of flow quota`)
-                await flowInstanceService.delete({ projectId: data.projectId, flowId: data.flowVersion.flowId })
-            }
-            else {
-                captureException(e)
-            }
+        const consumers: Record<ExecutionType, (job: Job) => Promise<void>> = {
+            [ExecutionType.BEGIN]: consumeRepeatingJob,
+            [ExecutionType.RESUME]: consumeDelayedJob,
         }
 
-        logger.info(`[repeatableJobConsumer] done job.id=${job.name} job.type=${job.data.triggerType}`)
+        const consumer = consumers[job.data.executionType]
+
+        await consumer(job)
     },
     {
         connection: createRedisClient(),
         concurrency: system.getNumber(SystemProp.FLOW_WORKER_CONCURRENCY) ?? 10,
     },
 )
 
-const consumePieceTrigger = async (data: RepeatableJobData): Promise<void> => {
+const consumeDelayedJob = async (job: Job<DelayedJobData, void, string>): Promise<void> => {
+    logger.warn(`[FlowQueueConsumer#consumeDelayedJob] flowRunId=${job.data.runId}`)
+    logger.debug(job, '[FlowQueueConsumer#consumeDelayedJob] job')
+
+    const { data } = job
+
+    await flowRunService.start({
```

Comment:
This might throw TASK_QUOTA_EXCEEDED, you can share handling part between two functions (consumePieceTrigger & consumeRepeatingJob)

Reply:
ouch! good catch!